updates/ryan_datierungVonBisUpdate.sql

```diff
@@ -0,0 +1,76 @@
+
+UPDATE einzelbeleg
+SET
+    VonTag = IF(VonTag IS NULL OR VonTag = '', 0, VonTag),
```

Comment:
This is definitely wrong.

`VonTag = IF(VonTag IS NULL OR VonTag = '', 0, VonTag)`

The Default value for VonTag in the Database is NULL because these are values, so this is the value that should be set in the migration script as well as when the form is saved if the field is empty (should be the same for all 4 date-related Von... and Bis... fields).

The logic for "Genauigkeit..." should be different because these are foreign keys that point to selektion_genauigkeit. and the default is "-1" which should be set in the migration script as well as whenever the form is saved.

Reply:
Das muss unbedingt Default 0 sein, 
![image](https://github.com/user-attachments/assets/dea5bbaf-e0f0-4563-bd57-200aad4a9885)
 und nicht wie hier -1

und hier NULL 
![image](https://github.com/user-attachments/assets/70199956-5f79-4899-aace-4091c15a97b5)

das ist auch so etwas schlampigk, einfach DEFAULT 0 schadet nicht auf irgendeine weise und macht das programm stabill

Reply:
sorry das erste bild ist falsch, ist Genauigkeit ... das ist ok das ander VonTag ist auch NULL aber wie gesagt ...

Reply:
Nein muss es nicht. Bei allen Selektions-Tabellen gilt die Konvention dass der Default -1 ist. Das sieht man wenn man sich die vorhandenen Werte z.B. in selektion_datgenauigkeit anschaut:

![image](https://github.com/user-attachments/assets/d285383a-e576-4101-9d4b-1cb712b8516f)

Der Default -1 bei Selektionen steckt an sehr vielen Stellen im Programmcode drin. 0 wäre speziell in diesem Fall "ca." und würde den Eintrag verfälschen. Der Optimale SOLL-Zustand wäre eigentlich dass es -1 nicht gäbe und der Defaut NULL wäre (so wie wir das bei allen anderen Fremdschlüsseln im Datenbereich haben aber nicht bei Selektionen) aber da es wie gesagt in sehr vielen Templates drinsteckt wurde die Konvention bisher beibehalten, sonst müssten wir das flächendeckend in der DB und in sämtlichen Templates ändern.